### ISSUE TITLE
onCreateMenu, node.path is null when an object without any key

### ISSUE THREAD

**Issue by hcsgzh** [josdejong/jsoneditor#698]
Hi there,

Thanks for this fantastic JSON editor! I need to use this JSON editor in one of my projects. 

I would like to have a customised menu for each different node based on the context of the node.

Everything is working good. However, I spot one small problem. For example.

```
json : {
        'array': [1, 2, 3],
        'boolean': true,
        'null': null,
        'number': 123,
        'object': {'a': 'b', 'c': 'd'},
        'string': 'Hello World',
        address: {}
    }
```

The address is an object and there is no default value in the object. when I click on the menu button. The onCreateMenu is evoked with two attributes, items, node. In this case, the node is {path: null}. Which is not what I expected. I would like to see {path: ["address", "null"]}.

![jsoneditor01](https://user-images.githubusercontent.com/6902389/57819970-e7fbe180-77de-11e9-96eb-73b7f742b869.PNG)

![jsoneditor02](https://user-images.githubusercontent.com/6902389/57819979-ef22ef80-77de-11e9-98ae-c3acc860e2ae.PNG)
#

Thanks for the lib again!

**Comment by josdejong** [comment]
Thanks for reporting @hcsgzh. 

The path should indeed be filled in. Not sure if `null` is a clear meaning to denote a not-yet-existing object property or array item though. Maybe we should return the path of the parent object, and an additional flag to denote that it's about adding something inside the object or array, like: `{path: ['address'], append: true}`. What do you think?

Besides the issue you encountered, I saw the `onCreateMenu` also has an issue when multiple nodes are selected.

**Comment by hcsgzh** [comment]
@josdejong ,

`{path: ['address'], append: true}`, this will do the job. 

I have another thought of the customised menu of the JSON Editor. I would like to update the templates in the onCreateMenu. This will make a customised menu to insert particular objects to a block. What do you think?

**Comment by josdejong** [comment]
ok clear. 

Yes sounds like a good idea to be able to customize templates depending on the path!

**Comment by josdejong** [comment]
Fixed via a9abad2cd499d0d2c108b1921d299904f93815fb (Fix not yet published)

I've extended the node object a bit differently, to also accomodate for multi selections:

```
{
  type: 'single' | 'multiple' | 'append'
  path: Array,
  paths: Array with paths
}
```

**Comment by josdejong** [comment]
This issue should be fixed now in `v5.33.0`
